pkg/registry/generic/rest/proxy.go

```diff
@@ -174,23 +174,53 @@ func (h *UpgradeAwareProxyHandler) tryUpgrade(w http.ResponseWriter, req *http.R
 func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 	dialAddr := netutil.CanonicalAddr(h.Location)
 
+	var dialer func(network, addr string) (net.Conn, error)
```

Comment:
Per tls [documentation](https://golang.org/pkg/crypto/tls/#Dial):
> Dial connects to the given network address using net.Dial and then initiates a TLS handshake, returning the resulting TLS connection. Dial interprets a nil configuration as equivalent to the zero configuration; see the documentation of Config for the defaults.

I don't know whether that code path is actually ever hit. I would expect we either have a transport with a valid tlsConfig or we are in the http case.